### PR TITLE
 bugfix: failed to validate namespace value

### PR DESCRIPTION
**Pull request by Ace-Tang**
### Ⅰ. Describe what this PR did

namespace related parameters not get validated, user should know
the valid value and invalid value when they specify the related
parameters.
remove repeated code in network namespace

### Ⅲ. Why don't you add test cases (unit test/integration test)? (你真的觉得不需要加测试吗？)

add namespace related tests for run command

### CODE REVIEW

File: daemon/mgr/container_validation.go [hunk 1]
```diff
@@ -278,3 +283,24 @@ func validateNvidiaDevice(r *types.Resources) error {
...
+func validateNSValue(hostConfig *types.HostConfig) error {
+	if !ns.Valid(specs.PIDNamespace, hostConfig.PidMode) {
+		return fmt.Errorf("invalid pid namespace mode %s", hostConfig.PidMode)
+	}
+
+	if !ns.Valid(specs.PIDNamespace, hostConfig.UTSMode) {
+		return fmt.Errorf("invalid uts namespace mode %s", hostConfig.UTSMode)
+	}
+
+	if !ns.Valid(specs.PIDNamespace, hostConfig.IpcMode) {
```
[the remark this answers is not in the document]
Comment (Ace-Tang) [reply daemon/mgr/container_validation.go]: Big Bug, thanks!

**Comment by Ace-Tang** [comment]
related to issue #2378

**Comment by Ace-Tang** [comment]
@rudyfly ， could you take a look at network part ?

File: daemon/mgr/container.go [hunk 2]
```diff
@@ -27,6 +27,7 @@ import (
 	"github.com/alibaba/pouch/pkg/errtypes"
 	"github.com/alibaba/pouch/pkg/meta"
 	mountutils "github.com/alibaba/pouch/pkg/mount"
+	ns "github.com/alibaba/pouch/pkg/namespace"
```
Comment (rudyfly) [inline daemon/mgr/container.go]: package named `namespace` is not suitable, you just move the container's network functions into the common package, in my opinion, named `net` is better， but I don't agree move then into pkg.

Comment (Ace-Tang) [reply daemon/mgr/container.go]: network function is one part of namespace, and some function are repeated, so make them as a pkg, wdyt?

File: daemon/mgr/network_utils.go [hunk 3]
```diff
@@ -2,49 +2,12 @@ package mgr
...
 )
 
-// IsContainer is used to check if network mode is container mode.
-func IsContainer(mode string) bool {
-	parts := strings.SplitN(mode, ":", 2)
-	return len(parts) > 1 && parts[0] == "container"
-}
-
-// IsHost is used to check if network mode is host mode.
-func IsHost(mode string) bool {
```
Comment (Ace-Tang) [inline daemon/mgr/network_utils.go]: Like these function is repeat with pid or user or .. namespace, this is not suitable in a network_util.go

**Comment by Ace-Tang** [comment]
I should disscuss with @rudyfly  before rebase it.

**Comment by fuweid** [comment]
ping @rudyfly

File: pkg/namespace/namespace.go [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+			return false
+		}
+
+	case specs.IPCNamespace:
+		return IsEmpty(mode) || IsNone(mode) || IsPrivate(ns, mode) || IsHost(mode) || IsShareable(mode) || IsContainer(mode)
+	}
+	return true
+}
+
+// IsEmpty indicates whether namespace mode is empty.
```
Comment (rudyfly) [inline pkg/namespace/namespace.go]: These functions aren't namespace's functions, so I don't agree the package named `namespace`.  These just check the container's network mode.

**Comment by fuweid** [comment]
please reopen it if you need

**Pull request closed without merging**